core/controllers/reader.py

```diff
@@ -421,6 +421,66 @@ class AnswerSubmittedEventHandler(base.BaseHandler):
     """Tracks a learner submitting an answer."""
 
     REQUIRE_PAYLOAD_CSRF_CHECK = False
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.ENTITY_ID_REGEX
+                }]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'POST': {
+            'session_id': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'old_state_name': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'answer': {
+                'schema': {
+                    'type': 'basestring'
+                }
+            },
+            'client_time_spent_in_secs': {
+                'schema': {
+                    'type': 'int'
+                }
+            },
+            'answer_group_index': {
+                'schema': {
+                    'type': 'int'
+                }
+            },
+            'rule_spec_index': {
+                'schema': {
+                    'type': 'int'
+                }
+            },
+            'version': {
+                'schema': {
+                    'type': 'int',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        'min_value': 1
+                    }]
+                },
+                'default_value': None
```

Comment:
Why `None`?

Reply:
Because if the params don't contain the version (which is a mandatory argument to us), setting it to None would raise InvalidInputException in the handler which is the desired behavior.

Reply:
But then if you remove the default value from here the schema validation will automatically raise if version is not set and you can remove the code below.

Reply:
Yes, that is certainly true but the test case written in reader_test.py (test_submit_answer_for_exploration_raises_error_with_no_version) is also expecting the error message to be "NONE EXP VERSION: Answer Submit" which is raised only if the control gets to the function. By removing the default_value: None, the control wouldn't reach there and the exception won't have the same message.